yt_dlp/extractor/instagram.py

```diff
@@ -28,7 +28,8 @@
 
 def _pk_to_id(media_id):
     """Source: https://stackoverflow.com/questions/24437823/getting-instagram-post-url-from-media-id"""
-    return encode_base_n(int(media_id.split('_')[0]), table=_ENCODING_CHARS)
+    pk = int(media_id.split('_')[0]) if type(media_id) is str else media_id
+    return encode_base_n(pk, table=_ENCODING_CHARS)
 
 
 def _id_to_pk(shortcode):
```

Comment:
Pretty sure `_id_to_pk` actually needs to be this or else it will convert incorrectly for private shortcodes from years ago
```diff
 def _id_to_pk(shortcode):
-    """Covert a shortcode to a numeric value"""
+    """Convert a shortcode to a numeric value"""
-    return decode_base_n(shortcode[:11], table=_ENCODING_CHARS)
+    return decode_base_n(shortcode[:-28], table=_ENCODING_CHARS)
```

Reply:
Do you not want to do this in this PR?

Reply:
Happy to add it here too, was not sure if it was relevant to this PR or just an observation